implementations/rust/ockam/ockam_node/src/storage/database/sqlx_from_row_types.rs

```diff
@@ -0,0 +1,80 @@
+use sqlx::database::HasValueRef;
+use sqlx::error::BoxDynError;
+use sqlx::postgres::any::AnyTypeInfoKind;
+use sqlx::{Any, Database, Decode, Type, Value, ValueRef};
+
+/// This type is used to map Option<T> fields for the types deriving `FromRow`
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nullable<T>(Option<T>);
+
+impl<T: Clone> Nullable<T> {
+    /// Create a new Nullable value
+    pub fn new(t: Option<T>) -> Self {
+        Nullable(t)
+    }
+
+    /// Return the Option corresponding to this value in the database
+    pub fn to_option(&self) -> Option<T> {
```

Comment:
Should we avoid cloning and use `into()` instead?

Reply:
I think we should address this is in a separate PR, because there would still be some cloning involved. There are many places where we do some cloning of individual fields to re-assemble them in the final desired type, to go from `&MyTypeRow` to `MyType`. What we gain in exchange is separate methods to parse the row fields if only a subset are needed.
We can definitely optimize this pattern but let's tackle this separately.